### PR TITLE
add test_negf files

### PR DESCRIPTION
**Pull request by AsymmetryChou**
Following the workflow of NEGF.py, I add a series of test_negf_xxxx.py files in /dptb/tests using 1D carbon chain as input.

- **test_negf_CFR.py**: check **ozaki_residues**
- **test_negf_density_Ozaki.py**: check class **Density** and **Ozaki's method** to calculate electron density
- **test_negf_Device.py**: check class **Device** and class **Lead** to calculate self-energy and green function
- **test_negf_hamiltonian**: check class **Hamiltonian**, including Hamiltonian Matrix calculation and output.
- **test_negf_run.py**: check the whole workflow of negf module with 1D carbon chain(12 atoms) as input.
- **test_negf_RGF.py**: check the ability of RGF.py to obtain Green Function by Matrix Inversion Operation. However at the current stage, RGF.py only does normal matrix inversion process rather than High Efficient one due to the lack of Hamiltonian's block tri-diagnal form

### CODE REVIEW

File: dptb/tests/data/postrun/run_config.json
Comment (AsymmetryChou) [inline dptb/tests/data/postrun/run_config.json:1]: Sorry! As I have run `pytest ./dptb/tests` in my PC, here the file paths change.  Maybe you could just save the former version of this file.

Comment (AsymmetryChou) [reply dptb/tests/data/postrun/run_config.json:1]: In the following files, there are the same problems. Just keep the old version's content~

File: dptb/tests/data/test_all/fancy_ones/checkpoint/best_nnsk_b4.000_c4.000_w0.300.json
Comment (floatingCatty) [inline dptb/tests/data/test_all/fancy_ones/checkpoint/best_nnsk_b4.000_c4.000_w0.300.json:1]: Why parameters in this file changed? will this affect other uni-test?

Comment (AsymmetryChou) [reply dptb/tests/data/test_all/fancy_ones/checkpoint/best_nnsk_b4.000_c4.000_w0.300.json:1]: It also bothers me.
It's a result coming from `pytest ./dptb/tests` in my PC.  Actucally the test files I add would not change this file. I think it 's better to keep the old version.

**Comment by AsymmetryChou** [comment]
I have fixed the problems above. Please check it.

**Pull request merged**